lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -1099,6 +1097,26 @@ ConstExprFunctionState::computeSingleStoreAddressValue(SILValue addr) {
       }
     }
 
+    if (auto *cai = dyn_cast<CopyAddrInst>(user)) {
```

Comment:
sorry I lost track of whether this chunk of code helps with the test case at hand.

I noticed 4 CopyAddr insts in the output of deabstraction pass, and they seem to be still present in the input to partition pass (when -O is on). Also they are not on tensor values. one example is:
```
copy_addr %147 to [initialization] %152 : $*Int64 // id: %153
```
(If this chunk helps with other test cases, including it in the patch is fine.)

Reply:
This is necessary to see through to the constants that get folded.